openfe/protocols/openmm_utils/multistate_analysis.py

```diff
@@ -0,0 +1,619 @@
+              from state i to state j.
+        """
+        # Get replica mixing statistics
+        mixing_stats = self.analyzer.generate_mixing_statistics()
+        transition_matrix = {'eigenvalues': mixing_stats.eigenvalues,
+                             'matrix': mixing_stats.transition_matrix}
+        return transition_matrix
+
+    @property
+    def replica_states(self):
+        """
+        Timeseries of states for each replica.
+        """
+        return self._replica_states
+
+    @property
+    def equilibration_iterations(self):
+        """
+        Number of iterations discarded as equilibration.
+        """
+        return self._equil_iters
+
+    @property
+    def production_iterations(self):
+        """
+        Number of production iterations from which energies are sampled.
+        """
+        return self._prod_iters
+
+    @property
+    def free_energy(self):
+        """
+        The free energy estimate from decorrelated unbiased samples
+        """
+        return self._free_energy
+
+    @property
+    def free_energy_error(self):
+        """
+        The MBAR estimate of the free energy estimate
+        """
+        return self._free_energy_err
+
+    @property
+    def forward_and_reverse_free_energies(self):
+        """
+        The dictionary forward and reverse analysis of the free energies
+        using the number of samples defined at class initialization
+        """
+        return self._forward_reverse
+
+    @property
+    def free_energy_overlaps(self):
+        """
+        A dictionary containing the estimated overlap matrix and corresponding
+        eigenvalues and scalars of the free energies.
+        """
+        return self._overlap_matrix
+
+    @property
+    def replica_exchange_statistics(self):
+        """
+        A dictionary containing the estimated replica exchange matrix
+        and corresponding eigenvalues.
+        """
+        if hasattr(self, '_exchange_matrix'):
+            return self._exchange_matrix
+        else:
+            errmsg = ("Exchange matrix was not generated, this is likely "
+                      f"{self.sampling_method} is not repex.")
+            raise ValueError(errmsg)
+
+    @property
+    def _unit_results_dict(self):
+        results_dict = {
+            'unit_estimate': self.free_energy,
+            'unit_estimate_error': self.free_energy_error,
+            '_unit_mbar_overlap': self.free_energy_overlaps,
+            '_forward_and_reverse_energies': self.forward_and_reverse_free_energies,
+            '_production_iterations': self.production_iterations,
+            '_equilibration_iterations': self.equilibration_iterations,
+            '_replica_states': self.replica_states}
+
+        if hasattr(self, '_exchange_matrix'):
+            results_dict['_replica_exchange_statistics'] = self.replica_exchange_statistics
+
+        return results_dict
+
+    def __del__(self):
+        self.close()
```

Comment:
this isn't required, the analyzer already does this

Reply:
removed